src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -268,16 +279,15 @@ static File makeChunkFile(File baseDir, int i)
   public static class Outer implements SmooshedWriter
   {
     private final int fileNum;
-    private final OutputStream out;
     private final int maxLength;
+    private final GatheringByteChannel channel;
 
-    private boolean open = true;
     private int currOffset = 0;
 
-    Outer(int fileNum, OutputStream out, int maxLength)
+    Outer(int fileNum, FileChannel channel, int maxLength)
     {
       this.fileNum = fileNum;
-      this.out = out;
+      this.channel = new BufferedWritableByteChannel(channel);
```

Comment:
@navis if you don't want to benchmark druid's code that uses this (what I understand, because it's tedious), I'm pro removing buffering, because it will allow to remove the hand-written buffering channel class. We can return to this only if we will see perf regression in druid, when we try to upgrade it to a version of java-util, which has this change.